sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -9004,4 +9021,25 @@ pi_result _pi_buffer::free() {
   return PI_SUCCESS;
 }
 
+inline pi_result piDeviceTime::get(uint64_t *deviceTime) {
+  if (!initialized) {
+    std::unique_lock{mutex};
+    initialized = true;
+    ZeTimerResolution = device->ZeDeviceProperties->timerResolution;
+    TimestampMaxCount =
+        ((1ULL << device->ZeDeviceProperties->kernelTimestampValidBits) - 1ULL);
+  }
```

Comment:
this seems to repeat machinery of existing `ZeCache`, can't it be reused?
or maybe you don't need that because ZeDeviceProperties is already "cached"

Reply:
Now using ZeDeviceProperties  directly